docs/FAQ.rst

```diff
@@ -11,44 +11,15 @@ LightGBM FAQ
 
 ------
 
-Critical Issues
```

Comment:
These changes are included in this PR because I originally came to this page to add @borchero to the list of maintainers.

Explanation of the changes to these docs:

* none of this applies to only "Critical" issues, so removed language like that
* list of maintainers was seriously out-of-date
    - many of these people have not interacted with the repo in years
    - some of the notes about specialization were wrong (e.g. implying that I only work on R and Dask)
* example questions don't add much information above what we ask for in the issue templates (https://github.com/microsoft/LightGBM/tree/master/.github/ISSUE_TEMPLATE)

Reply:
I'm basically proposing changing this from "list of all maintainers" to "who you should `@` if you need timely help".